extensions/binding/org.eclipse.smarthome.binding.weatherunderground/src/main/java/org/eclipse/smarthome/binding/weatherunderground/handler/WeatherUndergroundHandler.java

```diff
@@ -337,6 +337,11 @@ private WeatherUndergroundJsonData getWeatherData(Set<String> features, String l
             error = "Error parsing Weather Underground response";
             errorDetail = e.getMessage();
             statusDescr = "@text/offline.comm-error-parsing-response";
+        } catch (IllegalArgumentException e) {
+            // catch Illegal character in path at index XX: http://api.wunderground.com/...
```

Comment:
It is pretty hard to say which line of the code block actually might throw this exception. Could that be moved closer to the code that it handles?